mne_bids/path.py

```diff
@@ -762,31 +840,47 @@ def _find_best_candidates(params, candidate_list):
     return best_candidates
 
 
-def _get_bids_fname_from_filesystem(*, bids_basename, bids_root, sub, ses,
-                                    kind):
+def _get_bids_fpath_from_filesystem(*, bids_basename, bids_root, sub, ses,
+                                    kind, extension):
     if kind is None:
         kind = _infer_kind(bids_basename=bids_basename, bids_root=bids_root,
                            sub=sub, ses=ses)
 
-    data_dir = make_bids_folders(subject=sub, session=ses, kind=kind,
-                                 make_dir=False)
+    # support files above the bids_root/**/<kind>/ level
+    if kind == 'scans':
```

Comment:
this is code smell to me. What's going on?

Reply:
Yes, so this is an abuse of the `kind` kwarg. We should discuss this.

Reply:
So here, I'm using `kind` as the last `_` delimiter on the filename before the extension. This encompasses more then just `ieeg`, `eeg`, and `meg`. Most of these files can sit in the `<kind>` directory, but `*_scans.tsv` file possibly sits at a higher level on the `<session>` directory.

We could do some sort of like... globbing to find the files instead?, since technically `*_scans.tsv` file can sit even higher at the `<subject>` directory.

Reply:
> So here, I'm using `kind` as the last `_` delimiter on the filename before the extension. This encompasses more then just `ieeg`, `eeg`, and `meg`. Most of these files can sit in the `<kind>` directory, but `*_scans.tsv` file possibly sits at a higher level on the `<session>` directory.

Yep, I got that. But it shows that we're probably in need of an additional level of abstraction somehow, since `scans`, for example, is not a "kind" in the BIDS sense, if I understand correctly.

Reply:
Instead of `kind` kwarg, should we use `fkind`, or `file_kind`?

Reply:
@hoechenberger just bumping to get your thoughts?

Reply:
I believe this is resolved now, right?